Canvas expansion (1x1 and 2x2) / additional image editing options

Javier-bat opened the pull request:
---

### Canvas expansion (1x1 and 2x2)

A new button was added to select multiple coordinates, with the option of 2 (1x1) or 4 (2x2) coordinates.
This allows creating larger images or multiple contiguous areas.

<img width="1312" height="751" alt="image" src="https://github.com/user-attachments/assets/ea663f27-6ab1-4480-84c8-5afc81c79ad7" />

**Example of application point:**

<img width="777" height="108" alt="image" src="https://github.com/user-attachments/assets/8e2ea28f-1d88-49db-9d1d-1d1969fb667a" />


https://github.com/user-attachments/assets/f33a1d9d-1682-4bc6-834c-04bee4c7ca40



The canvas is saved so you don’t need to re-enter the coordinates when reloading the page.
The same painting logic is preserved.

---

### Additional image editing options

Three new image settings have been added:

1. **Transparency threshold**
   Removes pixels within the selected transparency threshold.
   Useful to clean up blurry edges on low-quality PNG images.


https://github.com/user-attachments/assets/54e5faeb-3a4a-4f12-b5e4-19a7c8edaca0



2. **Color algorithm**
   Allows choosing the method for approximating image colors to the available palette. Options:

   * **RGB (fast):** Euclidean distance in RGB space.
   * **Lab (CIE76):** Color difference in perceptual Lab space.
   * **CIEDE2000:** More accurate (but slower) perceptual calculation.
     The code selects the algorithm and finds the closest palette color using the chosen metric.

The RGB algorithm, being a linear mathematical calculation, can be quite inaccurate with a limited palette. To solve this, I added three alternative algorithms, allowing you to choose the most suitable one for each case.

https://github.com/user-attachments/assets/c1a08038-dadf-4746-b67a-04c7a04beb97



3. **Dithering (optional)**
   Applies error diffusion to simulate intermediate colors and reduce banding when using a limited palette.

   * **Enabled:** Distributes quantization error across neighboring pixels in a serpentine pattern.
   * **Disabled:** Simply maps to the nearest color without error diffusion.

**Recommendation:** Disable for cartoons; enable for more realistic pixel art.


https://github.com/user-attachments/assets/02c73fb8-1f92-41d1-86d9-121429b6065f



---

xacter commented:
When dithering, you need to use the same algorithm as you are currently using, because it gives a different image for people who are currently drawing, and the image ratio decreases.

Javier-bat commented:
> When dithering, you need to use the same algorithm as you are currently using, because it gives a different image for people who are currently drawing, and the image ratio decreases.

Reverted to the previous dithering algorithm. 
Do you have an image ready to test with?

Javier-bat commented:
> It looks similar, but it still reduces my work from 65% to 35%.

fixed. 

- Added a new color algorithm (OKLab), previously used, now set as default.

- Synced drawing order with the canvas.

- Se arreglaron bugs en el dibujado en modo mosaico
